.azure-pipelines/testscripts_analyse/analyse_testcases.py

```diff
@@ -0,0 +1,186 @@
+from constant import DATAPLANE_FEATURES, PR_TOPOLOGY_TYPE, PR_TOPOLOGY_MAPPING
+from report_data_storage import KustoConnector
+
+
+def topo_name_to_type(topo_name):
+    pattern = re.compile(r'^(wan|t0|t1|ptf|fullmesh|dualtor|t2|tgen|mgmttor|m0|mc0|mx|dpu)')
+    match = pattern.match(topo_name)
+    if match is None:
+        logging.warning("Unsupported testbed type - {}".format(topo_name))
+        return "unsupported"
+    topo_type = match.group()
+    if topo_type in ['mgmttor', 'dualtor']:
+        # certain testbed types are in 't0' category with different names.
+        topo_type = 't0'
+    if topo_type in ['mc0']:
+        topo_type = 'm0'
+    return topo_type
+
+
+def collect_all_scripts():
+    '''
+    This function collects all test scripts under the folder 'tests/'
+    and get the topology type marked in the script
+
+    The return value is a dict contains the script name and topology type
+    [{
+        'testscript': 'acl/custom_acl_table/test_custom_acl_table.py',
+        'topology': 't0'
+    }]
+    '''
+    location = sys.argv[1]
+
+    # Recursively find all files starting with "test_" and ending with ".py"
+    # Note: The full path and name of files are stored in a list named "files"
+    files = []
+    for root, dirs, file in os.walk(location):
+        for f in file:
+            if f.startswith("test_") and f.endswith(".py"):
+                files.append(os.path.join(root, f))
+    files = natsorted(files)
+
+    # Open each file and search for regex pattern
+    pattern = re.compile(r"[^@]pytest\.mark\.topology\(([^\)]*)\)")
+    test_scripts = []
+
+    for f in files:
+        # Remove prefix from file name:
+        filename = f[len(location) + 1:]
+        try:
+            with open(f, 'r') as file:
+                for line in file:
+                    # Get topology type of script from mark `pytest.mark.topology`
+                    match = pattern.search(line)
+                    if match:
+                        for topology in match.group(1).split(","):
+                            topology_mark = topology.strip().strip('"').strip('\'')
+                            result = {
+                                "testscript": filename,
+                                "topology": topo_name_to_type(topology_mark)
+                            }
+                            test_scripts.append(result)
+        except Exception as e:
+            logging.error('Failed to load file {}, error {}'.format(f, e))
+
+    return test_scripts
+
+
+def get_PRChecker_scripts():
+    '''
+    Check if a script is included in the PR checker for the corresponding topology type
+    '''
+    pr_test_scripts_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../", "pr_test_scripts.yaml")
+
+    # Get all the scripts included in different PR checker
+    try:
+        with open(pr_test_scripts_file) as f:
+            pr_test_scripts = yaml.safe_load(f)
+    except Exception as e:
```

Comment:
If the file open/read fails, `pr_test_scripts` will not be defined. The code will hit `NameError` in LINE#120.
Suggest not supress the exception here.

Reply:
Make sense, I will remove the try...except here.

Reply:
How about explicitly catch NameError and log accordingly. And use another clause to catch rest of the exceptions.

Reply:
I think we can set `pr_test_scripts={}` before this try...catch block, thus we can catch the error while opening file and avoid potential NameError